### PR TITLE
Use standard error messages, where obvious

### PR DESCRIPTION
**Pull request by bmeike**
Feature CBL-441

### CODE REVIEW

**Comment by bmeike** [comment]
mmm... some of the automated tests are failing.  Oddly they fail only from the CLI, not when run from AS.  Looking into it.

**Comment by bmeike** [comment]
All tests now passing.  Previous fix worked only for Android tests.  Pure Java tests failed (on NPE) for reasons that we ought to examine at some point.
Also improved tests so that they are less brittle.

**Comment by bmeike** [comment]
NVM... Broke the Java tests again.  Lemme do this again after I finish my coffee...

**Comment by bmeike** [comment]
All tests pass... even Java
The android tests will fail, though, until the companion push from couchbase-lite-android

**Comment by bmeike** [comment]
Just pushed couchbase-lite-android.  All tests should pass on Jenkins, when this push goes through.

File: lib/src/shared/test/java/com/couchbase/lite/LogTest.java [hunk 1]
```diff
@@ -478,6 +464,85 @@ public void testNonASCII() throws CouchbaseLiteException {
...
+    @Test
+    public void testFormatStandardMessage() {
+        Map<String, String> stdErr = new HashMap<>();
+        stdErr.put("FOO", "TEST DEBUG %2$s %1$d %3$.2f");
+        try {
+            Log.initLogging(stdErr);
+            assertEquals("TEST DEBUG arg 1 3.00", Log.formatStandardMessage("FOO", 1, "arg", 3.0F));
+        }
+        finally {
+            reloadStandardErrorMessages();
```
Comment (pasin) [inline lib/src/shared/test/java/com/couchbase/lite/LogTest.java]: Why do we need to reload the error messages after finishing the test? Doesn't the error message happen in CouchbaseLite.init()?

Comment (bmeike) [reply lib/src/shared/test/java/com/couchbase/lite/LogTest.java]: Yes it does.  But CouchbaseLite.init happens only once per application.

**Pull request merged**